ecommerce/api.py

```diff
@@ -162,3 +166,40 @@ def make_reference_id(order):
             A reference number for use with CyberSource to keep track of orders
     """
     return "MM-{}-{}".format(settings.CYBERSOURCE_REFERENCE_PREFIX, order.id)
+
+
+def get_new_order_by_reference_number(reference_number):
```

Comment:
the name `get_new_order_by_reference_number` seems inaccurate. it's supposed to refer to an existing order, isn't it?